File: generated-code/php/codecraft/TcpReadWrite/Model/RepairAction.php
<?php

namespace Model {
    require_once 'Stream.php';

    /**
     * Repair action
     */
    class RepairAction
    {
        /**
         * Target entity's ID
         */
        public int $target;
    
        function __construct(int $target)
        {
            $this->target = $target;
        }
    
        /**
         * Read RepairAction from input stream
         */
        public static function readFrom(\InputStream $stream): RepairAction
        {
            $target = $stream->readInt32();
            return new RepairAction($target);
        }
        
        /**
         * Write RepairAction to output stream
         */
        public function writeTo(\OutputStream $stream): void
        {
            $stream->writeInt32($this->target);
        }
    }
}

File: generated-code/pascal/codecraft/TcpReadWrite/Model/RepairAction.pas
unit RepairAction;

{$mode delphi}{$H+}

interface

uses
    Stream,
    SysUtils;

type
    // Repair action
    TRepairAction = class
        // Target entity's ID
        target: Int32;
        constructor Create(target: Int32);
        // Read RepairAction from input stream
        class function ReadFrom(stream: TStream): TRepairAction; static;
        // Write RepairAction to output stream
        procedure WriteTo(stream: TStream);
        function ToString: ansistring; override;
    end;

implementation

constructor TRepairAction.Create(target: Int32);
begin
    self.target := target;
end;

class function TRepairAction.ReadFrom(stream: TStream): TRepairAction;
var target: Int32;
begin
    target := stream.ReadInt32;
    result := TRepairAction.Create(target);
end;

procedure TRepairAction.WriteTo(stream: TStream);
begin
    stream.WriteInt32(target);
end;

function TRepairAction.ToString: ansistring;
begin
    result := 'RepairAction {';
    result += 'target=';
    result += IntToStr(target);
    result += '}';
end;

end.